[FX] Put tensor metadata into a NamedTuple in ShapeProp

jamesr66a opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#55930 [FX] Put tensor metadata into a NamedTuple in ShapeProp**

This makes things better-organized and less dict-ly typed

Differential Revision: [D27741730](https://our.internmc.facebook.com/intern/diff/D27741730)

Commit:
[FX] Put tensor metadata into a NamedTuple in ShapeProp

[ghstack-poisoned]

Commit:
Update on "[FX] Put tensor metadata into a NamedTuple in ShapeProp"


This makes things better-organized and less dict-ly typed

Differential Revision: [D27741730](https://our.internmc.facebook.com/intern/diff/D27741730)

[ghstack-poisoned]

ansley commented on torch/fx/passes/shape_prop.py:
```diff
@@ -1,7 +1,61 @@
+    dtype : torch.dtype
+    stride : Tuple[int]
+    memory_format : Optional[torch.memory_format]
+
```
Extra empty line?

Chillee commented on test/test_fx.py:
```diff
@@ -1054,8 +1054,8 @@ def forward(self, x):
             if node.op == 'output':
-                output_shape = node.args[0].meta['shape']
-                output_stride = node.args[0].meta['stride']
+                output_shape = node.args[0].meta['tensor_meta'].shape
```
Perhaps `tensor_meta` is redundant considering its on an attribute called `.meta`. Maybe just `meta['tensor']`?

jamesr66a replied:
`meta['tensor']` can be interpreted as containing a tensor, I think `tensor_meta` is the right level of descriptiveness

Commit:
Update on "[FX] Put tensor metadata into a NamedTuple in ShapeProp"


This makes things better-organized and less dict-ly typed

Differential Revision: [D27741730](https://our.internmc.facebook.com/intern/diff/D27741730)

[ghstack-poisoned]

Commit:
Update on "[FX] Put tensor metadata into a NamedTuple in ShapeProp"


This makes things better-organized and less dict-ly typed

Differential Revision: [D27741730](https://our.internmc.facebook.com/intern/diff/D27741730)

[ghstack-poisoned]
